core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -486,10 +486,12 @@ private[chisel3] class DynamicContext(
   val sourceRoots:         Seq[File],
   val defaultNamespace:    Option[Namespace],
   // Definitions from other scopes in the same elaboration, use allDefinitions below
-  val loggerOptions: LoggerOptions,
-  val definitions:   ArrayBuffer[Definition[_]],
-  val contextCache:  BuilderContextCache,
-  val layerMap:      Map[layer.Layer, layer.Layer]
+  val loggerOptions:               LoggerOptions,
+  val definitions:                 ArrayBuffer[Definition[_]],
+  val contextCache:                BuilderContextCache,
+  val layerMap:                    Map[layer.Layer, layer.Layer],
+  val includeInlineTestsForModule: Seq[String],
+  val includeInlineTestsWithName:  Seq[String]
```

Comment:
```suggestion
  val includeInlineTestsForModule: Seq[PathMatcher],
  val includeInlineTestsWithName:  Seq[PathMatcher]
```
These should be `PathMatchers`

Reply:
On second thought, maybe these should be collapsed into a single type:
```scala
class InlineTestIncluder(...) {
  def includeTestsForModule(name: String): Boolean = ???
  def includeTestsWithName(name: String): Boolean = ???
}
```

The reason I'm suggesting this is a code organization suggestion. It seems to me that the default behavior should be to include no tests, but what if the user passes only `--include-tests-module=Foo` or only passes `--include-tests-name=smoke*`. It seems to me that when only 1 argument is passed, then the other should default to `*`. How to express this logic is tricky, but it may be easiest to do by encapsulating the logic in a single class that is created from `Seq[Annotation]` so that it can consume all of the annotations created from the command-line and figure out what to do.

It would also let us hide the `PathMatcher` logic because it is a little bit weird feeling but it implements globs correctly so it's good to use--but probably good to hide.

Reply:
I agree that this seems like the most sensible default behavior.